functions/src/stories.ts

```diff
@@ -52,42 +51,94 @@ export const indexStories = functions
           title,
           description,
           tags,
-          likes: translation.metaData.likes,
-        } as storyRecord,
-        {autoGenerateObjectIDIfNotExist: true}
-        );
+        } as storyRecord;
+        if (change.before.exists) {
+          const objectID = await index.search("", {
+            filters: "translationID:" + translationID,
+            hitsPerPage: 1,
+          }).then((r) => r.hits[0]?.objectID);
+          await index.partialUpdateObject(
+              {objectID, ...entry},
+              {createIfNotExists: true}
+          );
+        } else {
+          await index.saveObject(
+              entry,
+              {autoGenerateObjectIDIfNotExist: true}
+          );
+        }
+      }
+    });
+
+export const updateStoryMeta = functions
+    .region("europe-central2")
+    .firestore.document(
+        "stories/{storyID}/translations/{translationID}"
+    )
+    .onWrite(async (change, context) => {
+      if (change.after.exists) {
+        const meta = change.after.data()?.metaData;
+        const likes = meta.likes ?? 0;
+        const views = meta.views ?? 0;
+
+        if (change.before.exists) {
+          const {_likes, _views} = change.before.data()?.metaData;
+          if (_likes === likes && _views === views) return;
+        }
+
+        const objectID = await index.search("", {
+          filters: "translationID:" + context.params.translationID,
+          hitsPerPage: 1,
+        }).then((r) => r.hits[0]?.objectID);
+        if (!objectID) return;
+
+        await index.partialUpdateObject({objectID, likes, views});
       }
     });
 
-// TODO
+export const indexTrendingStories = functions
+    .region("europe-central2")
+    .pubsub.schedule("every 72 hours")
+    .onRun(async () => {
+      await index.browseObjects({
+        query: "",
+        attributesToRetrieve: [
+          "storyID",
+          "translationID",
+        ],
+        batch: async (batch) => {
+          for (const hit of batch) {
+            const {storyID, translationID, objectID} = (hit as never);
```

Comment:
`never`? It's better to use the generic type argument in `index.browseObjects<storyRecord>`